packages/@aws-cdk-testing/cli-integ/lib/shell.ts

```diff
@@ -60,27 +60,42 @@ export async function shell(command: string[], options: ShellOptions = {}): Prom
       if (interaction) {
 
         if (interaction.prompt.test(lastLine.get())) {
+
           // subprocess expects a user input now.
-          // we have to write the input AFTER the child has started
-          // reading, so we do this with a small delay.
+          // first, shift the interactions to ensure the same interaction is not reused
+          remainingInteractions.shift();
+
+          // then, reset the last line to prevent repeated matches caused by tty echoing
+          lastLine.reset();
+
+          // now write the input with a slight delay to ensure
+          // the child process has already started reading.
           setTimeout(() => {
             child.writeStdin(interaction.input + (interaction.end ?? os.EOL));
-            remainingInteractions.shift();
           }, 500);
         }
 
       }
 
     });
 
-    child.onStderr(chunk => {
-      if (show === 'always') {
-        writeToOutputs(chunk.toString('utf-8'));
-      }
-      if (options.captureStderr ?? true) {
-        stderr.push(chunk);
-      }
-    });
+    if (tty && options.captureStderr === false) {
+      // in a tty stderr goes to the same fd as stdout
+      throw new Error(`Cannot disable 'captureStderr' in tty`);
```

Comment:
Don't you mean: cannot ENABLE captureStderr?

Reply:
No, I do in fact mean DISABLE. Notice the condition: `options.captureStderr === false` (default is `true`).

In a TTY we can't **disable** capturing of `stderr` because it is captured as part of `stdout`.